homeassistant/components/iammeter/sensor.py

```diff
@@ -0,0 +1,130 @@
+"""Support for iammeter via local API."""
+import asyncio
+from datetime import timedelta
+import logging
+
+import async_timeout
+from iammeter import real_time_api
+from iammeter.power_meter import IamMeterError
+import voluptuous as vol
+
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_PORT
+from homeassistant.exceptions import PlatformNotReady
+from homeassistant.helpers import debounce
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_PORT = 80
+DEFAULT_DEVICE_NAME = "IamMeter"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_NAME, default=DEFAULT_DEVICE_NAME): cv.string,
```

Comment:
Do we need this name option, when the sensor has a name and the user can rename the entities since we implement unique id?

Reply:
yes,I need the 'name' option.
the 'name' make sensors more ‘recognizeble’ to user as I am using 'serial number' as unique id,it's not very friendly to show to user.

Reply:
Isn't the sensor_name recognizable?

Reply:
I guess it doesn't say which device it's coming from?

Reply:
Yes,user may have several iammeter devices,such as monitor their all current of house,current of kinchen etc.Devices with same sensor_name of "current" can't be recognized easily.
it will not to matter if user has one device only.

Reply:
Ok.